### PR TITLE
fix(修复配置历史分页异常): 配置历史分页异常，所有页都是全部记录

### PR DESCRIPTION
**Pull request by liu-xg**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

修改配置历史记录分页异常 #9825 

## Brief changelog

HistoryConfigInfoMapper新增page sql构造方法

## Verifying this change

1、默认连接derby数据库，配置新增超过10条记录，查看分页
2、连接mysql数据库，配置新增配置新增超过10条记录，查看分页

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by 985492783** [comment]
it looked like conflict with the #9822.

**Review by The-Gamer-01 — DISMISSED** [review]

File: plugin/datasource/src/main/java/com/alibaba/nacos/plugin/datasource/impl/derby/HistoryConfigInfoMapperByDerby.java [hunk 1]
```diff
@@ -45,19 +45,27 @@ public String findDeletedConfig() {
...
+                + "WHERE data_id = ? AND group_id = ? AND tenant_id = ? ORDER BY nid DESC  OFFSET " + offset + " ROWS FETCH NEXT " + limit + " ROWS ONLY";
+    }
+
     @Override
     public String detailPreviousConfigHistory() {
         return "SELECT nid,data_id,group_id,tenant_id,app_name,content,md5,src_user,src_ip,op_type,gmt_create,gmt_modified "
                 + "FROM his_config_info WHERE nid = (SELECT max(nid) FROM his_config_info WHERE id = ?)";
     }
-    
+
```
Comment (KomachiSion) [inline plugin/datasource/src/main/java/com/alibaba/nacos/plugin/datasource/impl/derby/HistoryConfigInfoMapperByDerby.java:68]: please use nacos code style to reformat codes

**Review by KomachiSion — CHANGES_REQUESTED** [review]

**Comment by liu-xg** [comment]
收到发自我的 iPhone在 2023年1月16日，10:01，杨翊 SionYang ***@***.***> 写道：﻿
@KomachiSion requested changes on this pull request.



In plugin/datasource/src/main/java/com/alibaba/nacos/plugin/datasource/impl/derby/HistoryConfigInfoMapperByDerby.java:
>      @Override
     public String detailPreviousConfigHistory() {
         return "SELECT nid,data_id,group_id,tenant_id,app_name,content,md5,src_user,src_ip,op_type,gmt_create,gmt_modified "
                 + "FROM his_config_info WHERE nid = (SELECT max(nid) FROM his_config_info WHERE id = ?)";
     }
-    
+

please use nacos code style to reformat codes

**Comment by liu-xg** [comment]
receive



At 2023-01-16 10:01:32, "杨翊 SionYang" ***@***.***> wrote:

@KomachiSion requested changes on this pull request.

In plugin/datasource/src/main/java/com/alibaba/nacos/plugin/datasource/impl/derby/HistoryConfigInfoMapperByDerby.java:

>      @Override
     public String detailPreviousConfigHistory() {
         return "SELECT nid,data_id,group_id,tenant_id,app_name,content,md5,src_user,src_ip,op_type,gmt_create,gmt_modified "
                 + "FROM his_config_info WHERE nid = (SELECT max(nid) FROM his_config_info WHERE id = ?)";
     }
-    
+


please use nacos code style to reformat codes

**Comment by KomachiSion** [comment]
conflict with others PR, please rebase new codes and fix the conflicts.

**Comment by KomachiSion** [comment]
@liu-xg Please sign your cla.

**Comment by liu-xg** [comment]
已签署cla

**Comment by liu-xg** [comment]
> @liu-xg Please sign your cla.

completed

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**